### ISSUE TITLE
Validation of XML Schema files and Examples

### ISSUE THREAD

**Issue by nob140** [opengeospatial/CityGML-3.0Encodings#60]
I have just downloaded the .xsd files and example .gml files from GitHub, put them to one folder, modified the xsd file path to the local, changed cityGMLBase.xsd to CityGML.xsd, and validated them by both FZKViewer and VS Code with XML extension.
However, validation was failure with many errors.
Maybe both .xsd and example files are not updated?

And is there any problem in your environment to use same name space http://www.opengis.net/citygml/3.0 both for core.xsd and CityGML.xsd?
In my environment, I should rename all http://www.opengis.net/citygml/3.0 to http://www.opengis.net/citygml/core/3.0 for core.xsd while using http://www.opengis.net/citygml/3.0 for CityGML.xsd.
Please let me know if you already have discussions to solve this issue.
Thanks!

**Comment by TatjanaKutzner** [comment]
The error is in the `schemaLocation` references in the test data sets. They still reference `cityGMLBase.xsd` instead of `core.xsd`. 
The name of the file `cityGMLBase.xsd` was changed to `core.xsd` some time ago, see issue #52.

The file `CityGML.xsd` is a base profile, i.e., the union of all CityGML modules. This profile exists in CityGML 2.0 (see chapter 7.2 in the specification) and was also created for CityGML 3.0.
We should mention this in the specification as well!

For validating the test data sets you will either use the profile file or the individual xsd files.
I validated all test data sets against the latest release, but I used FME and the CityGML.xsd file for validation. Thus, FME did not read the schemaLocation references and that's why I unfortunately did not update the paths in the test data sets.

I will do that.

**Comment by nob140** [comment]
Thanks! It is intersting to know the difference of tool's behavior!

One validation error by FZKViewer and VS Code for importing CityGML.xsd was as follows:
_TargetNamespace.1: Expecting namespace 'http://www.opengis.net/citygml/3.0', but the target namespace of the schema document is 'http://www.opengis.net/citygml/profiles/base/3.0'.xml(TargetNamespace.1)_

It also looks like namespace collision of `http://www.opengis.net/citygml/3.0` in `CityGML.xsd` and `core.xsd`.
The easiest way to avoid it would be to allign xmlns and targetnamespace such as follows:
CityGML.xsd: `xmlns="http://www.opengis.net/citygml/3.0" targetNamespace="http://www.opengis.net/citygml/3.0"`
core.xsd: `xmlns:core="http://www.opengis.net/citygml/core/3.0" targetNamespace="http://www.opengis.net/citygml/core/3.0"`

SInce it is in such last-minute, I really would like to know workaround and include its example to the Annex of Encoding specification!

**Comment by clausnagel** [comment]
@nob140, I just tested XML validation with the `CityGML.xsd` base profile and one of the example datasets. All works fine. Note that `CityGML.xsd` defines its own target namespace `http://www.opengis.net/citygml/profiles/base/3.0`. You must declare this namespace and use it as reference in the `xsi:schemaLocation` attribute.

For example:

```xml
<CityModel xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" 
 xsi:schemaLocation="http://www.opengis.net/citygml/profiles/base/3.0 Schema/CityGML.xsd" 
 xmlns:base="http://www.opengis.net/citygml/profiles/base/3.0"
 [...more namespace declarations required for the dataset...]
 >
```

Please download my example [here](https://github.com/opengeospatial/CityGML-3.0Encodings/files/8046320/base_profile_example.zip) and check whether it works with your tools.

Changing the target namespace of `CityGML.xsd` is not required, and I cannot recommend re-using the namespace of `core.xsd`.

**Comment by nob140** [comment]
I am very happy to know that the problem is only in the example dataset side!
Let's modify the examples later.
Thanks!

**Comment by TatjanaKutzner** [comment]
I just updated the schemaLocation in the test data sets.

**Comment by nob140** [comment]
@clausnagel I have tested your example data and summarized the [results](https://github.com/nob140/CityGML3_test).
All errors of VS Code was resolved by using modified sample data successfully!
But unfortunately, FZKViewer detected 204 errors and I could not resolved so far.

@TatjanaKutzner Thanks for updating the exammples! Validation results are same, OK for VS Code, failure for FZKViewer. 
I would report to KIT...
